Automattic/course
# Add course landing page

## Introduced changes
* 8 new patterns to compose the new course landing page
* Course landing page pattern
* New template to control a menu variation
* Menu variation that disables the stick and doesn't use a background color
* Support the current sensei message
* Script to move the sensei message to a place after the cover block when the  'Page using transparent menu' is selected.

### Extra Notes: 
1) For some unknown reason, the new page pattern is not generating a preview, as soon we have a solution it will be handled in another PR.

### Testing instructions
Scenario 1: Pattern with the regular template
* Create a new course
* Follow the course creation wizard steps
* Select the `Course Landing Page with cover featured image`  pattern
* Check the layout

Scenario 2: Pattern using the Landing page 
* Create a new course
* Follow the course creation wizard steps
* Select the `Course Landing Page with cover featured image`  pattern
* Select the page template 'Page using transparent menu'
* Publish
* Check if the featured image is above the menu
* Scroll down the page and check that the menu stick is not enabled

Check the mobile version. 

## Videos

** Mobile Version**
[Screen Recording on 2022-10-23 at 11_29_59.webm](https://user-images.githubusercontent.com/38718/197397838-fb9ee4a2-8ef2-458a-b84a-51ff3a4e7d52.webm)

** Desktop Version**
[Screen Recording on 2022-10-23 at 11_32_20.webm](https://user-images.githubusercontent.com/38718/197397929-18bcbff8-35f6-411f-976f-4ed6282df17f.webm)

** Mobile Version**
[Screen Recording on 2022-10-23 at 11_29_59.webm](https://user-images.githubusercontent.com/38718/197397838-fb9ee4a2-8ef2-458a-b84a-51ff3a4e7d52.webm)




## Screenshots

**Desktop Version without featured image:**
![Screenshot on 2022-10-23 at 11_18_12](https://user-images.githubusercontent.com/38718/197397240-f30f1dbe-807e-46c3-a9b0-8d1b67c466de.png)

**Desktop Version with featured image:**
![Screenshot on 2022-10-23 at 11_22_59](https://user-images.githubusercontent.com/38718/197397434-59f660c2-e8b9-452a-8224-eaf515e192c6.png)
